shipment_planning/models/shipment_planning.py

```diff
@@ -0,0 +1,170 @@
+                % "\n".join([s.name for s in not_confirmed_shipments])
+            )
+        return self.write({"state": "in_progress"})
+
+    def action_done(self):
+        not_in_progress_shipments = self.filtered(lambda s: s.state != "in_progress")
+        if not_in_progress_shipments:
+            raise UserError(
+                _("Following shipment plannings are not in progress:\n %s")
+                % "\n".join([s.name for s in not_in_progress_shipments])
+            )
+        return self.write({"state": "done"})
+
+    def action_cancel(self):
+        not_started_shipments = self.filtered(
+            lambda s: s.state not in ("confirmed", "in_progress")
+        )
+        if not_started_shipments:
+            raise UserError(
+                _("Following shipment plannings are not started:\n %s")
+                % "\n".join([s.name for s in not_started_shipments])
+            )
+        return self.write({"state": "cancel"})
+
+    def action_draft(self):
+        not_started_shipments = self.filtered(lambda s: s.state != "cancel")
+        if not_started_shipments:
+            raise UserError(
+                _("Following shipment plannings are not canceled:\n %s")
+                % "\n".join([s.name for s in not_started_shipments])
+            )
+        return self.write({"state": "draft"})
+
+    @api.depends("shipment_type", "warehouse_id", "date_from", "date_to")
+    def _compute_picking_to_plan_domain(self):
+        for planning in self:
+            domain = [
+                ("picking_type_code", "=", planning.shipment_type),
+                ("location_id", "child_of", planning.warehouse_id.view_location_id.id),
+                ("state", "in", ("waiting", "confirmed", "assigned")),
+            ]
+            if planning.date_from:
+                domain = expression.AND(
+                    [
+                        domain,
+                        [
+                            (
+                                "scheduled_date",
+                                ">=",
+                                fields.Datetime.to_string(planning.date_from),
+                            )
+                        ],
+                    ]
+                )
+            if planning.date_to:
+                domain = expression.AND(
+                    [
+                        domain,
+                        [
+                            (
+                                "scheduled_date",
+                                "<=",
+                                fields.Datetime.to_string(planning.date_to),
+                            )
+                        ],
+                    ]
+                )
+            planning.picking_to_plan_domain = json.dumps(domain)
+
+    @api.model
+    def create(self, vals):
+        defaults = self.default_get(["name", "shipment_type"])
+        sequence = self.env.ref("shipment_planning.shipment_planning_outgoing_sequence")
```

Comment:
suggestion make sequence configurable from stock.warehouse so we will rid of xml here and it will be possible to set different sequences to different stock